### PR TITLE
Clean up Valkyrie Resource Factory and add Value Mapper

### PR DESCRIPTION
**Pull request by no-reply**
This is a second batch of Valkyrie setup.

The factory is updated to handle typed data using the `Valkyrie::Mapper` interface; tests are added for numeric data, booleans, and symbols/tokens, as well as enumerables of URIs.

AF/Hyrax identifiers are shifted to use Valkyrie's `alternate_identifier` feature. This is motivated by Valkyrie adapters overwriting the `id` attribute silently. This can be problematic if we're counting on that `id` to match the one used by `ActiveFedora`, so the `alternate_id` gives us a safe alternative.

Support for defined reflections is also added to the factory, with preliminary testing for complex relationships.

@samvera/hyrax-code-reviewers

### CODE REVIEW

File: spec/wings/value_mapper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+require 'wings/value_mapper'
+
+RSpec.describe Wings::ValueMapper do
+  subject(:mapper) { described_class.for(value) }
+  let(:value)      { 'a value' }
+  let(:uri)        { RDF::URI('http://example.com/moomin') }
+
+  describe '.for' do
+    it 'returns a value mapper' do
+      expect(described_class.for(value)).to be_a described_class
```
Comment (cjcolvar) [inline spec/wings/value_mapper_spec.rb]: Do you think it makes sense to shorten this to `expect(mapper)`?

File: spec/wings/value_mapper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      it 'casts internal values' do
+        expect(mapper.result).to contain_exactly(*cast_values)
+      end
+    end
+
+    context 'with a URI value'
+    context 'with a blank node value'
+    context 'with a boolean value'
+    context 'with a date value'
+    context 'with a numeric value'
```
Comment (cjcolvar) [inline spec/wings/value_mapper_spec.rb]: Is it possible to fill out these contexts prior to merge?

File: spec/wings/resource_factory_spec.rb [hunk 3]
```diff
@@ -96,16 +99,82 @@
...
+                            publisher: work.publisher,
+                            related_url: work.related_url,
+                            source: work.source
+    end
+  end
+
+  context 'with relationship properties' do
+    let(:pcdm_object) { book }
+
+    class BookWithPages < ActiveFedora::Base
```
[the remark this answers is not in the document]
Comment (no-reply) [reply spec/wings/resource_factory_spec.rb]: we could make these anonymous classes with

```ruby
let(:book_class) do
  Class.new(ActiveFedora::Base) do
     # class body
  end
end
```

File: spec/wings/value_mapper_spec.rb (hunk 1 above)
Comment (no-reply) [reply spec/wings/value_mapper_spec.rb]: i like keeping the `.for` call explicit here, since it's what we're testing. `mapper` could change in the future to support other tests.

File: spec/wings/value_mapper_spec.rb (hunk 2 above)
Comment (no-reply) [reply spec/wings/value_mapper_spec.rb]: i'm not sure they all will pass, presently. these are added mostly to catalogue things we need to check up on. at the very least, i expect blank node handling to be unsatisfying as things stand.

File: spec/wings/resource_factory_spec.rb [hunk 4]
```diff
@@ -96,16 +99,82 @@
...
+    let(:attributes) do
+      {
+        title: ['fake title', 'fake title 2'],
+        contributor: ['user1'],
+        description: ['a description'],
+        pages: [page1, page2]
+      }
+    end
+
+    # TODO: extract to Valkyrie?
```
[the remark this answers is not in the document]
Comment (no-reply) [reply spec/wings/resource_factory_spec.rb]: thanks. will fix

File: spec/wings/resource_factory_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (no-reply) [reply spec/wings/resource_factory_spec.rb]: @elrayle thoughts?

File: spec/wings/resource_factory_spec.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (no-reply) [reply spec/wings/resource_factory_spec.rb]: we should probably actually change this to check the alternate id now!
